openedx/features/course_experience/utils.py

```diff
@@ -125,6 +125,22 @@ def mark_last_accessed(user, course_key, block):
                 # we'll use the last child.
                 block['children'][-1]['resume_block'] = True
 
+    def recurse_mark_scored(block):
+        """
+        Mark this block as 'scored' if any of its descendents are 'scored' (that is, 'has_score' and 'weight' > 0).
+        """
+        is_scored = block.get('has_score', False) and block.get('weight', 1) > 0
+        # Use a list comprehension to force the recursion over all children, rather than just stopping
+        # at the first child that is scored.
+        children_scored = any([recurse_mark_scored(child) for child in block.get('children', [])])
```

Comment:
is there a way to get both:
* full recursion of children so that the case where only a deep descendent is scored will still result in the block being marked as scored, AND
* early exit (short circuit as soon as one scored child is found)

for instance, can you pass a kill switch boolean into the function, to halt further recursion if we don't need it?

Reply:
We want to mark all of the descendants (mostly because I'm not sure where we're going to need this info on the front-end for future experimentation), so we have to call `recurse_mark_scored` on each of the children regardless.

Reply:
ok thanks